src/cmd/ksh93/tests/functions.sh

```diff
@@ -1249,5 +1249,10 @@ expect_status=0
 [[ $actual == "$expect" ]] ||
     err_exit "autoload function skipped dir test wrong output (expected $(printf %q "$expect"), got $(printf %q "$actual"))"
 
+# ======
+# When a function unsets itself, it should not fail to be unset
+$SHELL -c 'fn() { unset -f fn; true; }; fn' 2> /dev/null || err_exit 'unset of POSIX function in the calling stack fails'
+$SHELL -c 'function ftest { ftest2; }; function ftest2 { unset -f ftest; }; ftest' 2> /dev/null || err_exit 'unset of ksh function in the calling stack fails'
```

Comment:
These tests are not complete. They check that the crash is gone, but they don't actually check whether unsetting the function succeeded.

There are various ways of checking that, but the most reliable is probably the brute approach: set `PATH=/dev/null`, then try to run the function a second time with `2>/dev/null` and check that the exit status is 127 (not found).

I'd also lose the global `2>/dev/null` on these tests. If `unset -f` ever prints an error message, I would want to know about that.

Reply:
I have made the requested changes to the two regression tests. I have also added another test as well for another potential bug in POSIX functions (although the new test does use `2> /dev/null` as it is expected to return an error).

Reply:
Oops, I forgot to check the return status. One moment.

Reply:
The regression tests in this pull request (there are now four added) have been fully updated.